lang/java/avro/src/test/java/org/apache/avro/data/TestJodaTimeConversions.java

```diff
@@ -1,245 +0,0 @@
-    Assert.assertEquals("Round-trip conversion should work", nowInstant, roundTrip);
-
-    long May_28_2015_21_46_53_221_instant = 1432849613221L;
-    DateTime May_28_2015_21_46_53_221 = new DateTime(2015, 5, 28, 21, 46, 53, 221, DateTimeZone.UTC);
-
-    Assert.assertEquals("Known date should be correct", May_28_2015_21_46_53_221,
-        conversion.fromLong(May_28_2015_21_46_53_221_instant, TIMESTAMP_MILLIS_SCHEMA, LogicalTypes.timestampMillis()));
-    Assert.assertEquals("Known date should be correct", May_28_2015_21_46_53_221_instant,
-        (long) conversion.toLong(May_28_2015_21_46_53_221, TIMESTAMP_MILLIS_SCHEMA, LogicalTypes.timestampMillis()));
-  }
-
-  @Test
-  public void testTimestampMicrosConversion() throws Exception {
-    TimestampMicrosConversion conversion = new TimestampMicrosConversion();
-
-    long May_28_2015_21_46_53_221_843_instant = 1432849613221L * 1000 + 843;
-    DateTime May_28_2015_21_46_53_221 = new DateTime(2015, 5, 28, 21, 46, 53, 221, DateTimeZone.UTC);
-
-    Assert.assertEquals("Known date should be correct", May_28_2015_21_46_53_221, conversion
-        .fromLong(May_28_2015_21_46_53_221_843_instant, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros()));
-
-    try {
-      conversion.toLong(May_28_2015_21_46_53_221, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros());
-      Assert.fail("Should not convert DateTime to long");
-    } catch (UnsupportedOperationException e) {
-      // expected
-    }
-  }
-
-  @Test
-  public void testLossyTimestampMicrosConversion() throws Exception {
-    TimestampMicrosConversion conversion = new LossyTimestampMicrosConversion();
-    long nowInstant = new Date().getTime() * 1000 + 674; // add fake micros
-
-    DateTime now = conversion.fromLong(nowInstant, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros());
-    long roundTrip = conversion.toLong(now, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros());
-    Assert.assertEquals("Round-trip conversion should lose microseconds", dropMicros(nowInstant), roundTrip);
-
-    long May_28_2015_21_46_53_221_843_instant = 1432849613221L * 1000 + 843;
-    DateTime May_28_2015_21_46_53_221 = new DateTime(2015, 5, 28, 21, 46, 53, 221, DateTimeZone.UTC);
-
-    Assert.assertEquals("Known date should be correct", May_28_2015_21_46_53_221, conversion
-        .fromLong(May_28_2015_21_46_53_221_843_instant, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros()));
-    Assert.assertEquals("Known date should be correct", dropMicros(May_28_2015_21_46_53_221_843_instant),
-        (long) conversion.toLong(May_28_2015_21_46_53_221, TIMESTAMP_MICROS_SCHEMA, LogicalTypes.timestampMicros()));
-  }
-
-  /*
-   * model.addLogicalTypeConversion(new
-   * JodaTimeConversions.TimeMicrosConversion());
-   * model.addLogicalTypeConversion(new
-   * JodaTimeConversions.TimestampMicrosConversion());
-   */
-  @Test
-  public void testDynamicSchemaWithDateConversion() throws ClassNotFoundException {
```

Comment:
I recommend adding similar test cases to `TestTimeConversions` class now - I can't see any ReflectData tests there. Here are the cases I'm thinking of:
```
  @Test
  public void testDynamicSchemaWithDateConversion() throws ClassNotFoundException {
    Schema schema = getReflectedSchemaByName("java.time.LocalDate", new TimeConversions.DateConversion());
    Assert.assertEquals("Reflected schema should be logicalType date", DATE_SCHEMA, schema);
  }

  @Test
  public void testDynamicSchemaWithTimeConversion() throws ClassNotFoundException {
    Schema schema = getReflectedSchemaByName("java.time.LocalTime", new TimeConversions.TimeMillisConversion());
    Assert.assertEquals("Reflected schema should be logicalType timeMillis", TIME_MILLIS_SCHEMA, schema);
  }

  @Test
  public void testDynamicSchemaWithTimeMicrosConversion() throws ClassNotFoundException {
    Schema schema = getReflectedSchemaByName("java.time.LocalTime", new TimeConversions.TimeMicrosConversion());
    Assert.assertEquals("Reflected schema should be logicalType timeMicros", TIME_MICROS_SCHEMA, schema);
  }

  @Test
  public void testDynamicSchemaWithDateTimeConversion() throws ClassNotFoundException {
    Schema schema = getReflectedSchemaByName("java.time.Instant", new TimeConversions.TimestampMillisConversion());
    Assert.assertEquals("Reflected schema should be logicalType timestampMillis", TIMESTAMP_MILLIS_SCHEMA, schema);
  }

  @Test
  public void testDynamicSchemaWithDateTimeMicrosConversion() throws ClassNotFoundException {
    Schema schema = getReflectedSchemaByName("java.time.Instant", new TimeConversions.TimestampMicrosConversion());
    Assert.assertEquals("Reflected schema should be logicalType timestampMicros", TIMESTAMP_MICROS_SCHEMA, schema);
  }

  private Schema getReflectedSchemaByName(String className, Conversion<?> conversion) throws ClassNotFoundException {
    // one argument: a fully qualified class name
    Class<?> cls = Class.forName(className);

    // get the reflected schema for the given class
    ReflectData model = new ReflectData();
    model.addLogicalTypeConversion(conversion);
    return model.getSchema(cls);
  }
```

@Fokko what do you think?

Reply:
I've restored the above tests. The VCS is a bit messy because I've removed `TestJodaTimeConversions.java` and removed everything with `Jsr310` from the filenames.

Reply:
Great, thanks Fokko!